### ISSUE TITLE
The usage of variable alpha

### ISSUE THREAD

**Issue by Bearx666** [swt-user/DMPO#4]
Hello, thanks for you impressive paper of DMPO. After reading the codes of DMPO, I'm a little confused about it.

According to the file [run_dmpo.sh](https://github.com/swt-user/DMPO/blob/main/run_dmpo.sh), the training entrance is the file [train_dmpo_efficient.py](https://github.com/swt-user/DMPO/blob/main/fastchat/train/train_dmpo_efficient.py) and the DMPO trainer is in the file [dmpo_trainer_efficient.py](https://github.com/swt-user/DMPO/blob/main/fastchat/train/dmpo_trainer_efficient.py). 

After reading this file, I found the definition of the weight in the [line 908-930](https://github.com/swt-user/DMPO/blob/main/fastchat/train/dmpo_trainer_efficient.py), which is shown as follows:

`

       batch['chosen_coefficient'] = torch.ones_like(batch['chosen_labels']).type(torch.FloatTensor)
          for i, index_list in enumerate(batch['chosen_index']):
          num_round = len(index_list)//2
          for j in range(num_round):
              start = index_list[2*j]
              end = index_list[2*j+1]
              if self.gamma<1:
                  alpha = self.gamma**j * (1-self.gamma**(num_round-j)) /  (1-self.gamma**(num_round))
              elif self.gamma==1:
                  alpha = (num_round-j)/num_round
              batch['chosen_coefficient'][i][start:end] = alpha

        batch['rejected_coefficient'] = torch.ones_like(batch['rejected_labels']).type(torch.FloatTensor)
        for i, index_list in enumerate(batch['rejected_index']):
            num_round = len(index_list)//2
            for j in range(num_round):
                start = index_list[2*j]
                end = index_list[2*j+1]
                if self.gamma<1:
                    alpha = self.gamma**j * (1-self.gamma**(num_round-j)) /  (1-self.gamma**(num_round))
                elif self.gamma==1:
                    alpha = (num_round-j)/num_round
                batch['rejected_coefficient'][i][start:end] = alpha`
And the alpha is the weight added to the sample compared to DPO. However, I have not found any else use of this alpha or "chosen_coefficient" or "rejected_coefficient" in this respiratory and I don't know why is that. Should this alpha be used in the loss calculation or somewhere else?

Hope you can give me some suggestion on how it is used in the DMPO and I'll appreciate it a lot.

**Comment by swt-user** [comment]
You could refer to

**Comment by swt-user** [comment]
Thanks for your interest! 
We concatenated the coefficient in [line 721-722](https://github.com/swt-user/DMPO/blob/main/fastchat/train/dmpo_trainer_efficient.py) and ultilized them in [line 852-855](https://github.com/swt-user/DMPO/blob/main/fastchat/train/dmpo_trainer_efficient.py).

**Comment by Bearx666** [comment]
Got it! Thanks for your patient explanations.
